### PR TITLE
Weight unit not selected if don't scroll the wheel

### PR DESCRIPTION
**Pull request by akaktus**
[-] Added optional date to DeliveryViewModel
[-] Changed UnitPicker and DatePicker

### CODE REVIEW

File: o-fish-ios/Views/Components/DatePickerWithButton.swift [hunk 1]
```diff
@@ -10,39 +10,25 @@ import SwiftUI
 struct DatePickerWithButton: View {
 
     var selectButtonTitle: String = "Select"
-    @Binding private(set) var date: Date?
+    @State private var date = Date()
 
-    var selectButtonClicked: (() -> Void)?
+    var selectButtonClicked: ((Date) -> Void)
```
Comment (kojiba) [inline o-fish-ios/Views/Components/DatePickerWithButton.swift:15]: keep closure optional

File: o-fish-ios/Views/Components/DatePickerWithButton.swift [hunk 2]
```diff
@@ -10,39 +10,25 @@ import SwiftUI
...
-            set: {
-                self.date = $0
-        })
-    }
 }
 
 struct DatePickerWithButton_Previews: PreviewProvider {
     static var previews: some View {
-        DatePickerWithButton(date: .constant(Date()))
+        DatePickerWithButton {_ in}
```
Comment (kojiba) [inline o-fish-ios/Views/Components/DatePickerWithButton.swift:32]: remove closure

File: o-fish-ios/Views/ReportFlow/Catch/UnitPickerWithButton.swift [hunk 3]
```diff
@@ -10,19 +10,18 @@ import SwiftUI
 struct UnitPickerWithButton: View {
     typealias UnitSpecification = CatchViewModel.UnitSpecification
 
+    var selectButtonTitle: String = "Select"
+    var selectButtonClicked: ((UnitSpecification) -> Void)
+    @State private var unit: UnitSpecification = UnitSpecification.allCases[0]
```
Comment (kojiba) [inline o-fish-ios/Views/ReportFlow/Catch/UnitPickerWithButton.swift:15]: change to
@State private var unit = UnitSpecification.allCases.first ?? .notSelected

File: o-fish-ios/Views/ReportFlow/Catch/UnitPickerWithButton.swift [hunk 4]
```diff
@@ -10,19 +10,18 @@ import SwiftUI
 struct UnitPickerWithButton: View {
     typealias UnitSpecification = CatchViewModel.UnitSpecification
 
+    var selectButtonTitle: String = "Select"
+    var selectButtonClicked: ((UnitSpecification) -> Void)
```
Comment (kojiba) [inline o-fish-ios/Views/ReportFlow/Catch/UnitPickerWithButton.swift:14]: make closure optional

File: o-fish-ios/Views/ReportFlow/Catch/UnitPickerWithButton.swift [hunk 5]
```diff
@@ -32,24 +31,26 @@ struct UnitPickerWithButton: View {
...
+            set: { self.unit = UnitSpecification(rawValue: $0) ?? .notSelected }
+        )
+    }
+
 }
 
 struct UnitPickerWithButton_Previews: PreviewProvider {
     static var previews: some View {
-        UnitPickerWithButton(unit: .constant(.kilograms))
+        UnitPickerWithButton {_ in}
```
Comment (kojiba) [inline o-fish-ios/Views/ReportFlow/Catch/UnitPickerWithButton.swift:54]: remove closure

File: o-fish-ios/Views/ReportFlow/Vessel/VesselView.swift [hunk 6]
```diff
@@ -133,6 +133,7 @@ struct VesselView: View {
         self.vessel.homePort = self.initialVessel.homePort
         self.vessel.nationality = self.initialVessel.nationality
         self.showingPrefilledAlert = false
+        informationComplete = !vessel.isEmpty
```
Comment (kojiba) [inline o-fish-ios/Views/ReportFlow/Vessel/VesselView.swift:136]: is this related to PR?

Comment (akaktus) [reply o-fish-ios/Views/ReportFlow/Vessel/VesselView.swift:136]: This fixed bug with warning state, when you choose prefill vessel information (quick fix)

**Comment by akaktus** [comment]
My forked repository was de-attached. I couldn't reattach it, so I created new PR.
